### PR TITLE
Fix DataDog trace exporter

### PR DESCRIPTION
**Pull request by owais**
# Description

Fixed DataDog trace exporter

`ddtrace` package introduced some breaking changes that break the
DataDog trace exporter. This commit downgrades the dependency to last
know working version to fix the exporter.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by ericmustin** [comment]
@owais thanks for the ping here. Afaik, nothing wrong with keeping things pinned for now like you’ve done, but I’ll sync with @majorgreys this week on how we can address long term in dd-trace-py. Will followup when I have more.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased](https://github.com/open-telemetry/opentelemetry-python-contrib/compare/v0.19b0...HEAD)
 
 ### Changed
+- Downgrade DataDog exporter's `ddtrace` dependency so the exporter continues to work.
```
Comment (toumorokoshi) [inline CHANGELOG.md:10]: Would it be more accurate to say that the version range has been restricted? We've never prescribed a specific version.

**Review by toumorokoshi — CHANGES_REQUESTED** [review]
minor nit about the changelog, otherwise LGTM!

I think regardless of how ddtrace chooses to handle it, we should get this merged in so we can keep CI working and users unbroken.

File: CHANGELOG.md (hunk 1 above)
Comment (owais) [reply CHANGELOG.md]: Yes. Updated.

**Comment by owais** [comment]
@ericmustin IMO direct dependencies like this should always specify a fixed version instead of an open range. My recommendation would be to keep the dependency pinned and have the maintainer of opentelemetry-exporter-datadog update it to newer versions as they see fit.

**Review by toumorokoshi — APPROVED** [review]
LGTM, thanks!

**Review by codeboten — APPROVED** [review]

**Pull request merged**